Controller.php

```diff
@@ -113,9 +114,15 @@ private function checkNonce()
     {
         $nonce = Common::getRequestVar('form_nonce', '', 'string', $_POST);
 
-        if (Common::getRequestVar('choice', 'no') != 'yes' ||
-            !Nonce::verifyNonce('VisitorGenerator.generate', $nonce)
-        ) {
+        if (Common::getRequestVar('choice', 'no') != 'yes') {
+            $notification = new Notification(Piwik::translate('VisitorGenerator_ConfirmVisitorGeneration'));
+            $notification->context = Notification::CONTEXT_ERROR;
+            Notification\Manager::notify('confirmVisitorGeneration', $notification);
+
+            Piwik::redirectToModule('VisitorGenerator', 'index');
```

Comment:
Ideally we'd not do a redirect or an exit in a private method since it's hard to debug and unexpected but it's ok for now